ts/Stock/Indicators/VBP/VBPIndicator.ts

```diff
@@ -590,7 +591,19 @@ class VBPIndicator extends SMAIndicator {
             rangeStep: number,
             zoneStartsLength: number;
 
-        if (!lowRange || !highRange) {
+        // If the compare mode is set on the main series, change the VBP
+        // zones to fit new extremes, #16277.
+        const mainSeries = indicator.linkedParent;
+        if (
+            !indicator.options.compareToMain &&
+            mainSeries.options.compare &&
```

Comment:
Should this only be available for `compare`?

Reply:
Again, some rubish before the compare composition improvements. Thanks!